builtin/logical/aws/path_static_roles.go

```diff
@@ -0,0 +1,271 @@
+	if rawRotationPeriod, ok := data.GetOk(paramRotationPeriod); ok {
+		config.RotationPeriod = time.Duration(rawRotationPeriod.(int)) * time.Second
+
+		if err := b.validateRotationPeriod(config.RotationPeriod); err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, fmt.Errorf("missing '%s' parameter", paramRotationPeriod)
+	}
+
+	// Upsert role config
+	newRole, err := logical.StorageEntryJSON(formatRoleStoragePath(config.Name), config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal object to JSON: %w", err)
+	}
+	err = req.Storage.Put(ctx, newRole)
+	if err != nil {
+		return nil, fmt.Errorf("failed to save object in storage: %w", err)
+	}
+
+	// Bootstrap initial set of keys if they did not exist before. AWS Secret Access Keys can only be obtained on creation,
+	// so we need to boostrap new roles with a new initial set of keys to be able to serve valid credentials to Vault clients.
+	existingCreds, err := req.Storage.Get(ctx, formatCredsStoragePath(config.Name))
+	if err != nil {
+		return nil, fmt.Errorf("unable to verify if credentials already exist for role '%q': %w", config.Name, err)
+	}
+	if existingCreds == nil {
+		err := b.createCredential(ctx, req.Storage, config)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create new credentials for role '%q': %w", config.Name, err)
+		}
+
+		err = b.credRotationQueue.Push(&queue.Item{
+			Key:      config.Name,
+			Value:    config,
+			Priority: time.Now().Add(config.RotationPeriod).Unix(),
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to add item into the rotation queue for role '%q': %w", config.Name, err)
+		}
+	}
+
+	return &logical.Response{
+		Data: formatResponse(config),
+	}, nil
+}
+
+func (b *backend) pathStaticRolesDelete(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	roleName, ok := data.GetOk(paramRoleName)
+	if !ok {
+		return nil, fmt.Errorf("missing '%s' parameter", paramRoleName)
+	}
+
+	err := b.deleteCredential(ctx, req.Storage, roleName.(string))
+	if err != nil {
+		return nil, fmt.Errorf("failed to clean credentials while deleting role '%q': %w", roleName.(string), err)
+	}
+
+	return nil, req.Storage.Delete(ctx, formatRoleStoragePath(roleName.(string)))
+}
+
+func (b *backend) validateRoleName(name string) error {
+	if name == "" {
+		return errors.New("empty role name attribute given")
+	}
+	return nil
+}
+
+func (b *backend) validateIAMUserExists(ctx context.Context, req *logical.Request, username string) error {
+	c, err := b.clientIAM(ctx, req.Storage)
+	if err != nil {
+		return fmt.Errorf("unable to validate username %q: %w", username, err)
+	}
+
+	// we don't really care about the content of the result, just that it's not an error
+	out, err := c.GetUser(&iam.GetUserInput{
+		UserName: aws.String(username),
+	})
+	if err != nil || out.User == nil {
+		return fmt.Errorf("unable to validate username %q: %w", username, err)
+	}
+	if *out.User.UserName != username {
```

Comment:
since this is a static user and could potentially be modified out-of-band (i.e. not by Vault), I am wondering if this "friendly name" check is sufficient? Should we also be checking the  [unique identifier](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html#identifiers-unique-ids)?

Reply:
This raises a point probably worth discussing. In this particular case, I think we're fine since we're checking the username requested matches the username that comes back.

In general, though, we use usernames as our reference to the associated IAM user in this feature, which raises the possibility that changing a username could cause Vault to "lose track" of its associated IAM user?

Reply:
>  I think we're fine since we're checking the username requested matches the username that comes back

Could the username that comes back have a different unique ID though? And could this username that comes back have different IAM policies/permissions?

We could store the ID on a CreateOperation and always ensure that a given username is associated with the ID that we expect. We use a similar "trust-on-first-use" pattern in GitHub Auth.

Reply:
I think that would work - let the user supply a friendly name and store the invariant id for validation, with the trust exception for the creation case.